### PR TITLE
Added metadata definition for chat.postMessage and history

### PR DESCRIPTION
**Pull request by tchapuis**
The Slack documentation says that we can include metadata in `chat.postMessage` arguments. See : https://api.slack.com/metadata

OpenApi spec for this argument doesn't seem to be up to date.

So I added it to the patched version for message definition and `chat.postMessage` method. I also added the `include_all_metadata` argument for the `conversations.history` method

### CODE REVIEW

**Comment by yngc0der** [comment]
The temporary solution is create a custom endpoint class, and use this class to call api method:

```php
<?php

use JoliCode\Slack\Api\Endpoint\ChatPostMessage;

final class FixedChatPostMessageEndpoint extends ChatPostMessage
{
    protected function getFormOptionsResolver(): \Symfony\Component\OptionsResolver\OptionsResolver
    {
        $optionsResolver = parent::getFormOptionsResolver();
        $optionsResolver->setDefined(['metadata']);
        $optionsResolver->setAllowedTypes('metadata', ['string']);
        return $optionsResolver;
    }
}


$slack->executeEndpoint(new FixedChatPostMessageEndpoint([
    'channel' => $channel,
    'text' => $text,
    'metadata' => $metadata,
]));
```

File: resources/slack-openapi-sorted.patch [hunk 1]
```diff
@@ -26,6 +26,7 @@
                  "created",
                  "id",
                  "is_intro",
+\ No newline at end of file
```
Comment (damienalexandre) [inline resources/slack-openapi-sorted.patch:29]: **💭 thought:** ‏I think this should not appear in the patch file :thinking:

**Review by damienalexandre — CHANGES_REQUESTED** [review]
Hello, sorry for the very long delay on the review.

I fear the ".patch" file is wrong, it includes A LOT of changes that have nothing to do with the metadata definition :thinking: Maybe it's just because Slack updated their definition, but we should then split this in two PR.

Thank you for your understanding :pray:

**Comment by tchapuis** [comment]
Hello,
The patch file was a pain to get it right 😅

**Comment by damienalexandre** [comment]
Thanks for working on this!

Tests on `main` were red, so I fixed them in https://github.com/jolicode/slack-php-api/pull/162 ;
could you please:

- rm -rf generated/
- vendor/bin/jane-openapi generate --config-file=.jane-openapi.php

And push?

**Comment by damienalexandre** [comment]
Awesome, thank you so much!

**Comment by damienalexandre** [comment]
Version v4.6.0 is published!

**Pull request merged**